src/libfm/libfm.cpp

```diff
@@ -119,6 +119,11 @@ int main(int argc, char **argv) {
 		if (! cmdline.hasParameter(param_init_stdev)) { cmdline.setValue(param_init_stdev, "0.1"); }
 		if (! cmdline.hasParameter(param_dim)) { cmdline.setValue(param_dim, "1,1,8"); }
 
+		if (cmdline.hasParameter(param_save_model) && !cmdline.getValue(param_method).compare("mcmc")) {
+			std::cout << "**WARNING**: load/save enabled only for SGD and ALS. Nothing will be saved." << std::endl;
+			cmdline.setValue(param_save_model, "FALSE_MCMC_SAVE");
```

Comment:
A cleaner solution would be to remove the param_save_model entry from the cmdline-options. You would have to extend the Cmdline-Class by a removeParameter(const std::string& parameter) method. This would make the code in line 430 easier, because it wouldn't have to check for the magic string.

Even simpler would be to just throw an exception here. This would make the code even simpler and the user of the tool won't accidentally overlook the warning.